pkg/storage/stores/tsdb/manager.go

```diff
@@ -281,11 +287,47 @@ func (m *tsdbManager) BuildFromWALs(t time.Time, ids []WALIdentifier) (err error
 	return nil
 }
 
-func indexBuckets(from, through model.Time, tableRanges config.TableRanges) (res []string) {
+func migrateMultitenantDir(dir string, tableRange config.TableRange, logger log.Logger) error {
```

Comment:
Initially I was migrating WALs based on the ts, but removed this code given that the ts is the creation timestamp of the WAL and is not helpful when deciding which period it belongs to. I felt it would be okay not to migrate WALs as long as we gracefully shutdown ingesters during the migration which takes care of [building the prev head](https://github.com/grafana/loki/blob/ashwanth/index-shipper-multi-store/pkg/storage/stores/tsdb/head_manager.go#L220). 
 
But if we want to retain the WALs during migration, we could move them to the latest period dir and have that instance of head manager process them, wdyt?